source/Hardpoint.h

```diff
@@ -42,11 +49,14 @@ class Hardpoint {
 	const Angle &GetAngle() const;
 	// Get the base angle that this weapon is aimed at (without harmonization/convergence), relative to the ship.
 	const Angle &GetBaseAngle() const;
+	// Get the movable angle. Return value is invalid if this is omnidirectional.
+	std::pair<Angle, Angle> GetMovableAngle() const;
```

Comment:
See my comment below about renaming to turnRange.
I would also expect omnidirectional turrets to get a sensible default return-value like pair(Angle(0.), Angle(0.)).

Reply:
I prefer to return a value pair(Angle(baseAngle+180), Angle(baseAngle+180)), but anyway I'm not sure the use-case.

Technically speaking, the 360 degree arc is different from omnidirectional in the hardpoint. The latter can rotate infinitely. I know it's trivial...

Reply:
Such differences can be quite relevant if a turret indeed can rotate 360 degrees but needs to cross the 1 point where it cannot turn (meaning that the turret needs a lot more in-game time to perform the turn).
Nice to differentiate between the 2 situations.

Reply:
Sorry, I'm wrong. 360 degree means 0 degree.
Anyway, it returns pair(Angle(baseAngle+180), Angle(baseAngle+180)) now.